### PR TITLE
[IRGen] Support movesAsLike for the array variant of @_rawLayout

### PR DESCRIPTION
**Pull request by Azoy**
This is similar to https://github.com/swiftlang/swift/pull/73955, but for the `likeArrayOf` variant. Note that all elements within the array _must_ be initialized.

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please test

**Review by rjmccall — APPROVED** [review]
LGTM

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please smoke test Linux

File: test/IRGen/raw_layout.swift [hunk 1]
```diff
@@ -241,32 +287,106 @@ entry(%0 : $*Cell<T>):
...
+//===----------------------------------------------------------------------===//
+// SmallVectorOf2MovesAsLike<T> initializeWithTake
+//===----------------------------------------------------------------------===//
+
+// CHECK-LABEL: define {{.*}} ptr @"$s10raw_layout25SmallVectorOf2MovesAsLikeVwtk"(ptr {{.*}} %dest, ptr {{.*}} %src, ptr %"SmallVectorOf2MovesAsLike<T>")
+// CHECK: [[T_ADDR:%.*]] = getelementptr inbounds ptr, ptr %"SmallVectorOf2MovesAsLike<T>", {{i64|i32}} 2
+// CHECK-NEXT: [[T:%.*]] = load ptr, ptr [[T_ADDR]]
+// CHECK: [[STRIDE_GEP:%.*]] = getelementptr inbounds %swift.vwtable, ptr {{%.*}}, i32 0, i32 9
+// CHECK-NEXT: [[STRIDE:%.*]] = load {{i64|i32}}, ptr [[STRIDE_GEP]]
+// CHECK: [[OFFSET_0:%.*]] = mul i64 0, [[STRIDE]]
```
Comment (kateinoigakukun) [inline test/IRGen/raw_layout.swift:339]: @Azoy Seems like the `mul` index operand types also need to be `{{i64|i32}}`
```
/home/build-user/build/buildbot_linux/wasmstdlib-linux-x86_64/test-wasi-wasm32/IRGen/Output/raw_layout.swift.tmp/raw_layout.sil:339:11: error: CHECK: expected string not found in input
// CHECK: [[OFFSET_0:%.*]] = mul i64 0, [[STRIDE]]
          ^
<stdin>:3260:28: note: scanning from here
 %stride = load i32, ptr %2, align 4, !invariant.load !5
                           ^
<stdin>:3260:28: note: with "STRIDE" equal to "%stride"
 %stride = load i32, ptr %2, align 4, !invariant.load !5
                           ^
<stdin>:3261:2: note: possible intended match here
 %3 = mul i32 0, %stride
```
https://ci.swift.org/job/oss-swift-pr-test-crosscompile-wasm-ubuntu-20_04/1253/console

**Pull request merged**